src/Extension/ExtensionManager.php

```diff
@@ -279,7 +283,7 @@ protected function unpublishAssets(Extension $extension)
      */
     public function getAsset(Extension $extension, $path)
     {
-        return $this->paths->public.'/assets/extensions/'.$extension->getId().$path;
+        return $this->assetsFilesystem->url($extension->getId()."/$path");
```

Comment:
I can't find where this method is used to properly tell, but aren't replacing a path with  a url here ?

Reply:
This isn't used. Frankly I'm not sure why we even have it. I removed it in the first version of this PR, but reverted that change as it was a bit too spontaneous.

If assets aren't always on disk, then the URL to assets won't always be a path. If we want to keep it, I'll update the docblock.

Reply:
I think it was used in very early versions of Flarum, today though I don't think that anyone uses it. I think that it would be safe to remove this.

Reply:
I don't think this can be removed, @askvortsov1 is onto something here.

In case you use an S3 bucket to store assets files to your public directory you need to be able to resolve the actual, full URL that the asset is published as to. Disks in Laravel support this functionality using the `url` method. 

However, I don't think any extension currently is using it and that might be a problem in itself 😂

Reply:
Let's tackle the promotion or redesign of this feature separately, then.